src/stores/DialogStore/DialogStore.ts

```diff
@@ -20,179 +59,74 @@ export class DialogStore {
-    @action showRegionDialog = () => {
-        this.regionDialogVisible = true;
-    };
-    @action hideRegionDialog = () => {
-        this.regionDialogVisible = false;
-    };
-
-    // Hotkey
-    @observable hotkeyDialogVisible: boolean;
-    @action showHotkeyDialog = () => {
-        this.hotkeyDialogVisible = true;
-    };
-    @action hideHotkeyDialog = () => {
-        this.hotkeyDialogVisible = false;
-    };
-
-    // About
-    @observable aboutDialogVisible: boolean;
-    @action showAboutDialog = () => {
-        this.aboutDialogVisible = true;
-    };
-    @action hideAboutDialog = () => {
-        this.aboutDialogVisible = false;
-    };
-
-    // Preference
-    @observable preferenceDialogVisible: boolean;
-    @action showPreferenceDialog = () => {
-        this.preferenceDialogVisible = true;
-    };
-    @action hidePreferenceDialog = () => {
-        this.preferenceDialogVisible = false;
-    };
-
-    // Layout
-    @observable saveLayoutDialogVisible: boolean;
-    @action showSaveLayoutDialog = (oldLayoutName?: string) => {
-        this.saveLayoutDialogVisible = true;
-        AppStore.Instance.layoutStore.setOldLayoutName(oldLayoutName);
-    };
-    @action hideSaveLayoutDialog = () => {
-        this.saveLayoutDialogVisible = false;
-    };
-
-    // Workspace
-    @observable workspaceDialogMode = WorkspaceDialogMode.Hidden;
-    @action showWorkspaceDialog = (mode = WorkspaceDialogMode.Save) => {
-        this.fileBrowserDialogVisible = false;
-        this.workspaceDialogMode = mode;
-    };
-    @action hideWorkspaceDialog = () => {
-        this.workspaceDialogMode = WorkspaceDialogMode.Hidden;
-    };
-
-    // Workspace sharing
-    @observable shareWorkspaceDialogVisible: boolean;
-    @action showShareWorkspaceDialog = () => {
-        this.shareWorkspaceDialogVisible = true;
-    };
-    @action hideShareWorkspaceDialog = () => {
-        this.shareWorkspaceDialogVisible = false;
+    zIndexManager = AppStore.Instance.zIndexManager;
+
+    @action showDialog = (id: string, options?: showDialogOptions) => {
+        if (id === DialogId.Snippet) {
+            if (options?.newSnippet) {
+                SnippetStore.Instance.clearActiveSnippet();
+            } else if (options?.snippet) {
+                SnippetStore.Instance.setActiveSnippet(options.snippet, options.name);
+            }
+
+            if (!this.dialogVisible.get(DialogId.Snippet)) {
+                this.dialogVisible.set(DialogId.Snippet, true);
+                this.zIndexManager.assignIndex(DialogId.Snippet);
+            }
+
+            this.zIndexManager.updateIndexOnSelect(id);
+        } else if (!this.dialogVisible.get(id)) {
+            switch (id) {
+                case DialogId.Layout:
+                    this.dialogVisible.set(DialogId.Layout, true);
+                    if (options?.oldLayoutName) {
+                        AppStore.Instance.layoutStore.setOldLayoutName(options.oldLayoutName);
```

Comment:
Do we need to set `oldLayoutName` as `undefined` if `options.oldLayoutName` is not provided?

Reply:
The default oldLayoutName is "". Is it not enough?

Reply:
I think we still call `setOldLayoutName` when `oldLayoutName` is not provided: https://github.com/CARTAvis/carta-frontend/pull/2307/files#diff-f1a1f1a428941f84abc390302a4a5bf6140d77ad098789d91f3f76b843e2b22eL63

Reply:
okay